### PR TITLE
[Merged by Bors] - Update for Rust 1.50

### PR DESCRIPTION
**Pull request by paulhauner**
## Issue Addressed

NA

## Proposed Changes

Rust 1.50 has landed :tada:

The shiny new `clippy` peers down upon us mere mortals with disgust. Brutish peasants wrapping our `usize`s in superfluous `Option`s... tsk tsk.

I've performed the goat sacrifice and corrected our evil ways in this PR. Tonight we shall pray that Github Actions bestows the almighty green tick upon us.

## Additional Info

NA

### CODE REVIEW

File: beacon_node/network/src/nat.rs [hunk 1]
```diff
@@ -52,11 +52,7 @@ pub fn construct_upnp_mappings<T: EthSpec>(
                 // Just use the first IP of the first interface that is not a loopback and not an
                 // ipv6 address.
                 if !interface.is_loopback() {
-                    if let IpAddr::V4(_) = interface.ip() {
-                        Some(interface.ip())
-                    } else {
-                        None
-                    }
+                    interface.ip().is_ipv4().then(|| interface.ip())
```
Comment (michaelsproul) [inline beacon_node/network/src/nat.rs:55]: Nice :sunglasses:

**Review by michaelsproul — APPROVED** [review]
LGTM!

**Comment by paulhauner** [comment]
I'll try an optimistic bors :crossed_fingers: 

bors r+

**Comment by paulhauner** [comment]
Seems CI hasn't updated to 1.50 yet :thinking: I guess I'll wait a few hours and try again...

**Comment by paulhauner** [comment]
bors retry

**Comment by michaelsproul** [comment]
finally.

**Pull request closed without merging**